src/python/WMCore/WMSpec/StdSpecs/PromptReco.py

```diff
@@ -61,7 +61,7 @@ def buildWorkload(self):
                                                splitAlgo = self.procJobSplitAlgo,
                                                splitArgs = self.procJobSplitArgs,
                                                stepType = cmsswStepType,
-                                               forceUnmerged = True)
+                                               forceUnmerged = ["write_ALCARECO"] if 'ALCARECO' in self.writeTiers else False)
```

Comment:
Yeah, exactly. Overloading the parameter so that the clients that use the boolean functionality don't need to care that you can also pass a list seemed the least invasive option to me. Makes the low level code a bit more complex, but only in one limited location.